### PR TITLE
feat(alerts): implement NerdGraph policy search

### PR DESCRIPTION
**Pull request by zlesnr**

### CODE REVIEW

File: pkg/alerts/policies.go [hunk 1]
```diff
@@ -193,6 +197,32 @@ func (a *Alerts) QueryPolicy(accountID, id int) (*QueryPolicy, error) {
 	return &resp.Actor.Account.Alerts.Policy, nil
 }
 
+// QueryPolicySearch searches NerdGraph for policies.
+func (a *Alerts) QueryPolicySearch(accountID int, params AlertsPoliciesSearchCriteriaInput) ([]QueryPolicy, error) {
```
Comment (sanderblue) [inline pkg/alerts/policies.go]: I think we've been returning an array of pointers - i.e. `[]*QueryPolicy` - for consistency

Comment (zlesnr) [reply pkg/alerts/policies.go]: Done.

File: pkg/alerts/types.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	* **policyName** - The alert policy name
+	* **conditionName** - The alert condition name
+	* **conditionType** - The alert condition type, such as app_metric
+	* **conditionRunbookUrl** - The alert condition's runbook url
+	* **product** - The target product (e.g., SYNTHETICS)
+	* **targetId** - The ID of the alerts target
+	* **targetName** - The name of the alerts target
+	* **nrqlEventType** - The NRQL event type
+	* **tag** - Arbitrary tags associated with some entity (e.g., FACET from a NRQL query)
+	* **nrqlQuery** - The NRQL query string */
```
Comment (sanderblue) [inline pkg/alerts/types.go]: Do these show up in godocs?

Comment (jthurman42) [reply pkg/alerts/types.go]: The probably don't as the first word in the comment is not the Name of the object.  We should refactor the typegen to resolve that as a follow up

Comment (zlesnr) [reply pkg/alerts/types.go]: Let's find out.

**Pull request merged**